File: src/main/java/de/weimarnetz/registrator/configuration/NetworksConfiguration.java
package de.weimarnetz.registrator.configuration;

import java.util.Map;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

import lombok.Data;

@ConfigurationProperties(prefix = "network")
@Component
@Data
public class NetworksConfiguration {

    private Map<String, NetworkInformation> map;
}

File: src/main/java/de/weimarnetz/registrator/configuration/NetworksConfiguration.kt
package de.weimarnetz.registrator.configuration

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "network")
data class NetworksConfiguration(
    val map: Map<String, NetworkInformation>
)

data class NetworkInformation(
    val maxNodeNumber: Int = 0,
    val minNodeNumber: Int = 0,
    val leaseDays: Int = 0
)